x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/advanced_step/advanced_step_form.tsx

```diff
@@ -481,18 +486,31 @@ export const AdvancedStepForm: FC<CreateAnalyticsStepProps> = ({
               }
             )}
           >
-            <EuiFieldText
-              placeholder={
-                jobType !== undefined
-                  ? DEFAULT_MODEL_MEMORY_LIMIT[jobType]
-                  : DEFAULT_MODEL_MEMORY_LIMIT.outlier_detection
-              }
-              disabled={isJobCreated}
-              value={modelMemoryLimit || ''}
-              onChange={(e) => setFormState({ modelMemoryLimit: e.target.value })}
-              isInvalid={modelMemoryLimitValidationResult !== null}
-              data-test-subj="mlAnalyticsCreateJobWizardModelMemoryInput"
-            />
+            <>
+              <EuiFieldText
+                placeholder={
+                  jobType !== undefined
+                    ? DEFAULT_MODEL_MEMORY_LIMIT[jobType]
+                    : DEFAULT_MODEL_MEMORY_LIMIT.outlier_detection
+                }
+                disabled={isJobCreated || useEstimatedMml}
+                value={useEstimatedMml ? estimatedModelMemoryLimit : modelMemoryLimit || ''}
+                onChange={(e) => setFormState({ modelMemoryLimit: e.target.value })}
+                isInvalid={modelMemoryLimitValidationResult !== null}
+                data-test-subj="mlAnalyticsCreateJobWizardModelMemoryInput"
+              />
+              <EuiSpacer size="xs" />
```

Comment:
Nit - `size s` might look better here, as with `xs` it looks a bit squished up to me when the switch has focus:
![image](https://user-images.githubusercontent.com/7405507/96579966-80996980-12cf-11eb-823c-aa3bef76b174.png)

Reply:
👍  Updated in d1f4ec267189bc2ff4a9b59fea3d64c0de2c8fba